spring-cloud-gcp-starters/spring-cloud-gcp-starter-core/src/main/java/org/springframework/cloud/gcp/core/autoconfig/GcpContextAutoConfiguration.java

```diff
@@ -55,22 +61,43 @@
 
 	private static final List<String> SCOPES_LIST = ImmutableList.of(PUBSUB_SCOPE, SQLADMIN_SCOPE);
 
+	private static final Log LOGGER = LogFactory.getLog(GcpContextAutoConfiguration.class);
+
 	@Autowired
 	private GcpProperties gcpProperties;
 
 	@Bean
 	@ConditionalOnMissingBean
 	public CredentialsProvider googleCredentials() throws Exception {
+		CredentialsProvider credentialsProvider;
+
 		if (!StringUtils.isEmpty(this.gcpProperties.getCredentialsLocation())) {
-			return FixedCredentialsProvider
+			credentialsProvider = FixedCredentialsProvider
 					.create(GoogleCredentials.fromStream(
 							this.gcpProperties.getCredentialsLocation().getInputStream())
 					.createScoped(SCOPES_LIST));
 		}
+		else {
+			credentialsProvider = GoogleCredentialsProvider.newBuilder()
+					.setScopesToApply(SCOPES_LIST)
+					.build();
+		}
+
+		Credentials credentials = credentialsProvider.getCredentials();
+
+		if (credentials instanceof UserCredentials) {
+			LOGGER.info("Default credentials provider for user "
+					+ ((UserCredentials) credentials).getClientId());
```

Comment:
everything less then `warn` should be wrapped to the `LOGGER.is...Enable()` to avoid extra computation for the string concatenation.